### ISSUE TITLE
WSL exits with code 1, Rancher Desktop GUI starting though.

### ISSUE THREAD

**Issue by Cliftonz** [rancher-sandbox/rancher-desktop#1538]
### Rancher Desktop Version

1.0.1

### Rancher Desktop K8s Version

NA

### Which container runtime are you using?

containerd (nerdctl)

### What operating system are you using?

Windows

### Operating System / Build Version

19044

### What CPU architecture are you using?

x64

### Linux only: what package format did you use to install Rancher Desktop?

N/A

### Windows User Only

Tailscale

### Actual Behavior

Rancher GUI start but wsl version does not.

### Steps to Reproduce

Start Rancher Desktop

### Result

[logs.zip](https://github.com/rancher-sandbox/rancher-desktop/files/8020089/logs.zip)
Rancher Desktop starts but the wsl distribution does not.

### Expected Behavior

WSL vm should start correctly.

**Comment by ericpromislow** [comment]
Here's the summary:

`background.log`:
```
2022-02-08T02:04:11.910Z: Kubernetes was unable to start: Error: wsl.exe exited with code 1
    at ChildProcess.<anonymous> (C:\Users\ZAC16\AppData\Local\Programs\Rancher Desktop\resources\app.asar\dist\app\background.js:1:8656)
    at ChildProcess.emit (node:events:394:28)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12) {
  stdout: ''
}
```

`wsl.log`:
```
2022-02-08T02:04:01.717Z: Running command wsl --distribution rancher-desktop --exec /mnt/c/Users/ZAC16/AppData/Local/Programs/Rancher Desktop/resources/resources/linux/wsl-helper k3s kubeconfig...
2022-02-08T02:04:11.892Z: WSL: executing: /mnt/c/Users/ZAC16/AppData/Local/Programs/Rancher Desktop/resources/resources/linux/wsl-helper k3s kubeconfig: Error: wsl.exe exited with code 1
```

`wsl-exec.log`:
```
 * Starting k3s ... [ ok ]
Error: Timed out waiting for k3s kubeconfig to exist
Usage:
  wsl-helper k3s kubeconfig [flags]

Flags:
  -h, --help               help for kubeconfig
      --k3sconfig string   Path to k3s kubeconfig (default "/etc/rancher/k3s/k3s.yaml")

Global Flags:
      --verbose count   enable extra logging

Error: Timed out waiting for k3s kubeconfig to exist
```

**Comment by ericpromislow** [comment]
At this point, I would try clearing the existing backend and restart.  With Rancher Desktop shut down:

In Powershell:
```
wslconfig /u rancher-desktop-data
wslconfig /u rancher-desktop
```

Then in either powershell or bash (I prefer bash in an Ubuntu WSL, being a foreign speaker in psland):
```
$ cd $HOME/AppData
$ rm -fr {Local,Roaming}/rancher-desktop/*
```

And then restart.

If you get the same problem, could you please log it here, and also `wsl` into `rancher-desktop` and see if `k3s` is running (`ps auxww | grep k3s` is good enough)?

**Comment by mook-as** [comment]
The logs from `wsl-exec.log` indicate that this may be resolved by #1516 — we had a ten-second time out; if it takes longer than that for k3s to start enough to write out the kubeconfig file, we would fail like this.

**Comment by Cliftonz** [comment]
This was resolved by updating to windows 11 to fix wsl 2 interface issue, then clearing the rancher data like eric suggested.

However, when the system worked. I was getting some very vague issues like the first and I just decided to change the port and that worked.

I would recommend looking to see if the port is already taken and giving the appropriate response to the user so they know they should use another port.

Thanks for the help!!!
